atef/widgets/config/find_replace.py

```diff
@@ -0,0 +1,1009 @@
+            self,
+            'File saved',
+            'File saved successfully, would you like to open this file?'
+        )
+
+        if reply == QtWidgets.QMessageBox.Yes:
+            self._window._new_tab(data=self.orig_file, filename=filename)
+
+    def apply_all(self) -> None:
+        self.prompt_apply()
+        self.update_title()
+
+    def prompt_apply(self) -> None:
+        # message box with details on remaining changes
+        self.update_change_list()
+        if len(self.all_actions) <= 0:
+            return
+
+        reply = QtWidgets.QMessageBox.question(
+            self,
+            'Apply remaning edits?',
+            (
+                'Would you like to apply the remaining '
+                f'({len(self.all_actions)}) edits?'
+            )
+        )
+        if reply == QtWidgets.QMessageBox.Yes:
+            for action in self.all_actions:
+                action.apply()
+
+            # clear all rows
+            self.edits_table.clearContents()
+            self.details_list.clear()
+
+    def update_title(self) -> None:
+        """
+        Update the title.  Will be the name and the number of unapplied edits
+        """
+        file_name = os.path.basename(self.fp)
+        if len(self.all_actions) > 0:
+            file_name += f'[{len(self.all_actions)}]'
+        self.file_name_label.setText(file_name)
+        self.type_label.setText(type(self.orig_file).__name__)
+
+    def update_change_list(self) -> None:
+        """
+        update the global change list, gathering all ``FindReplaceAction``'s
+        """
+        # walk through edits_table, gather list of list of paths
+        self.all_actions = []
+        for row_idx in range(self.edits_table.rowCount()):
+            template_widget = self.edits_table.cellWidget(row_idx, 0)
+            self.all_actions.extend(template_widget.get_actions())
+
+        self.update_title()
+
+    def show_changes_from_edit(self, *args, **kwargs) -> None:
+        """
+        Populate the details_list with each action from the selected edit.
+        Provide helpful text if the edit is incomplete or invalid.
+        """
+        self.details_list.clear()
+        # on selected callback, populate details table
+        selected_ranges = self.edits_table.selectedRanges()
+        if not selected_ranges:
+            return
+
+        edit_row_widget: TemplateEditRowWidget = self.edits_table.cellWidget(
+            selected_ranges[0].topRow(), 0
+        )
+
+        if not isinstance(edit_row_widget, TemplateEditRowWidget):
+            # placeholder text if nothing is selected
+            l_item = QtWidgets.QListWidgetItem("select an edit or click an edit's"
+                                               "refresh button to show details.")
+            return
+        elif not edit_row_widget.get_details_rows():
+            l_item = QtWidgets.QListWidgetItem(
+                'provide search and replace text to show details...'
```

Comment:
Hmm, it wasn't clear to me why the "edit details" weren't showing:
<img width="803" alt="image" src="https://github.com/pcdshub/atef/assets/5139267/beb52117-d9d6-439f-aed1-a1e21dc803c1">
I think this should probably read "Click on edit in the left list to see details; if nothing shows here there is no match" or something to that effect